### PR TITLE
Fix: `edit-comments-faster` appears on other users' comments in issues

### LINKED ISSUE #4712: `edit-comments-faster` appears on other users' comments in issues
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.

### Description

Open an issue, e.g.: https://github.com/facebook/jest/issues/6316 and scroll through the comments in the conversations tab. You will see edit buttons on all comments, including comments from other users. When you click on the edit button, you see an error message: `Sorry, something went wrong`.

### Console errors

A 403 Forbidden when the page attempts to retrieve the edit form for that comment:

```
XHR GET https://github.com/facebook/jest/issue_comments/.../edit_form?textarea_id=issuecomment-...-body&comment_context= [HTTP/2 403 Forbidden 321 ms]
```

### Example URL

https://github.com/facebook/jest/issues/6316 (locked issue)
https://github.com/facebook/jest/issues/11791 (regular issue)

### Browser(s) used

Firefox

### PR DESCRIPTION
**Pull request by gamemaker1**
## Related issues

Closes #4712 if merged.

## What this PR does

Does not add the edit button on comments from other users if the current user is not allowed to edit them.

## Test URLs

https://github.com/facebook/jest/issues/6316 (locked issue)
https://github.com/facebook/jest/issues/11791 (regular issue)

### CODE REVIEW

File: source/features/edit-comments-faster.tsx [hunk 1]
```diff
@@ -7,13 +7,13 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
-function canEditDiscussion(): boolean {
-	return pageDetect.isDiscussion() && select.exists('[title^="You are a maintainer"], [title^="You are a collaborator"]');
+function canEditComments(): boolean {
+	return select.exists('[aria-label^="You are the owner"], [title^="You are a maintainer"], [title^="You are a collaborator"]');
```
Comment (gamemaker1) [inline source/features/edit-comments-faster.tsx:11]: Not sure if the maintainer/collaborator checks work

Comment (darkred) [inline source/features/edit-comments-faster.tsx:11]: Based on https://github.com/sindresorhus/refined-github/pull/4415#issuecomment-850970454, you'll also need `[aria-label^="You have been invited to collaborate"]`, i.e.

```suggestion
	return select.exists('[aria-label^="You have been invited to collaborate"], [aria-label^="You are the owner"], [title^="You are a maintainer"], [title^="You are a collaborator"]');
```

for issues/PRs of repos you are a Collaborator (so you can edit all comments).

&nbsp;
Plus, there's still the uncovered case when you are the owner/maintainer/collaborator of a repo (so you can edit all comments), and there's an issue/PR/discussion where you haven't made a comment yet (so the checks don't apply), 
but you can edit the title (=even if you're not the author).

Comment (darkred) [reply source/features/edit-comments-faster.tsx:11]: Yes, the checks still work.

Comment (gamemaker1) [reply source/features/edit-comments-faster.tsx:11]: Will add that check (most probably by checking if the user can edit the title or a settings tab is present), thanks

**Comment by fregante** [comment]
This only works if the current user left a comment already. For example I don't see any edit buttons on this page.

Screen Shot

**Comment by fregante** [comment]
And of course now that I left this comment is shows up:

Screen Shot

**Comment by fregante** [comment]
> Did you try `js-previewable-comment-form previewable-comment-form write-selected` first?

**Comment by gamemaker1** [comment]
> Did you try `js-previewable-comment-form previewable-comment-form write-selected` first?

Yes, I couldn't find that selector on any of the issue pages (the jest issue, the issue this PR fixes) except on the text box where you enter a new comment.

**Comment by gamemaker1** [comment]
> And of course now that I left this comment is shows up:

Maybe we should search for a settings tab as well then? It most probably doesn't show up because those labels are present only on your comments.

**Comment by gamemaker1** [comment]
It works pretty well now

**Comment by gamemaker1** [comment]
Also, this file seems to have been renamed #4745 - should I rename the file here too or should I try opening a new PR and cherry-picking the commits?

**Comment by yakov116** [comment]
> Also, this file seems to have been renamed by #4745 - should I rename the file here too or should I try opening a new PR and cherry-picking the commits?

Just merge from main, no need for a new PR.

**Comment by fregante** [comment]
Feel free to open a new PR when it's ready for review. It's probably easier to update your branch and then make the changes to the file again.

**Pull request closed without merging**